### PR TITLE
feat: add courses endpoint and new data scrapping 

### PR DESCRIPTION
**Pull request by angga1518**
- create endpoint needed for ulas kelas (description, prerequisite, course_code, and curriculum)
- Scrape new data needed for ulas kelas

It looks likely I accidentally reformat the scapper/main.py file. So kindly check the differences from this link 
https://www.diffchecker.com/5VXgAaky

### for new scrape flow, kindly check the image below
image

### Response preview
```
{
  "courses": [
    {
      "code": "CSGE602012", 
      "credit": 3, 
      "curriculum": "06.00.12.01-2016", 
      "description": "", 
      "name": "Aljabar Linier", 
      "prerequisite": "", 
      "term": 3
    }, 
   {...}
   ]
}
```

thank you

### CODE REVIEW

**Comment by fahdikrie** [comment]
Hello, Kak Angga!

Sorry for the late review. I just tried the newly added `/courses` endpoint, and it works & integrates just fine with the existing database.

Oh, a bit of a nitpick but I notice that there are inconsitencies on the `description` & `prerequisite` field whereas it could either be blank or null for the 'empty' values. Not sure how thats gonna affect anything though, but consider this as a mere heads up to UlasKelas team 😂
Screen Shot 2021-09-15 at 20 09 54

Regarding the best practices, I don't have much experience on the backend so i'm gonna pass the review to Kak Dipta (cc: @hsjsjsj009).

Cheers!

**Review by fahdikrie — APPROVED** [review]

File: backend/app/services/scrapper/schedule_scrapper.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ def callback(ch, method, properties, body):
             period.last_update_at = now
             period.save()
             app.logger.info(f"Done scrapping kd_org: {method.routing_key}; period: {active_period}; at: {now} UTC")
+            generate_desc_prerequisite(period, username, password)
```
Comment (hsjsjsj009) [inline backend/app/services/scrapper/schedule_scrapper.py:50]: @angga1518 Can you give brief documentation about this function?

File: backend/app/views/main.py [hunk 2]
```diff
@@ -139,3 +139,16 @@ def scrap_all_schedule():
         password=password
     )
     return jsonify(response), status_code
+
+@router_main.route('/courses', methods=['GET'])
+def get_all_courses():
```
Comment (hsjsjsj009) [inline backend/app/views/main.py:144]: @angga1518 Can you give the response preview of this endpoint in the pull request description?

**Review by hsjsjsj009 — COMMENTED** [review]
@angga1518 Have some comments

File: backend/app/services/scrapper/schedule_scrapper.py (hunk 1 above)
Comment (angga1518) [reply backend/app/services/scrapper/schedule_scrapper.py:51]: updated

File: backend/app/views/main.py (hunk 2 above)
Comment (angga1518) [reply backend/app/views/main.py:144]: updated

**Pull request merged**